R/doeResponseSurfaceMethodology.R

```diff
@@ -787,26 +812,113 @@ doeResponseSurfaceMethodology <- function(jaspResults, dataset, options, ...){
+
+  if (length(options[["rsmMax"]]) + length(options[["rsmMin"]]) + length(options[["rsmTar"]]) == 1)
+    desire_final <- desire_final[[1]][[1]]
+
+  op  <- data.frame(x1 = seq(-1,1,0.25))
+  op1 <- length(options[["rsmVariables"]])
+
+  if (op1 > 1) {
+    for (i in 2:op1) {
+      op <- cbind(op, seq(-1,1,0.25))
+    }
+    names(op) <- paste0("x",1:op1)
   }
-  # print(unique_l)
-  comb <- expand.grid(unique_l)
-  # print(names(data))
-  unique_l2 <- list()
-  for (i in 1:(ncol(data)-1)) {
-    unique_l2[[i]] <- unique(data[,i])
+
+
+  possibleCodedCombinationsOfPredictors <- expand.grid(op)
+
+  rsmOpt_2 <- function(predictorCombinationFunction, rsm, dObject, space = "square") {
+    conv <- numeric(length(rsm))
+    for (j in seq_along(rsm)) {
+      conv[j] <- predict(rsm[[j]], newdata = possibleCodedCombinationsOfPredictors[i,])
+    }
+    out <- do.call(predict, list(object = dObject, newobject = conv))
+
+    if (space == "circular") {
+      if (sqrt(sum(possibleCodedCombinationsOfPredictors[i,]^2)) > 1.682)
+        out <- 0
+    }
+    else if (space == "square")
+      if (any(abs(possibleCodedCombinationsOfPredictors[i,]) > 1.682))
+        out <- 0
+    out
   }
-  for (i in 1:length(unique_l)) {
-    for (j in 1:length(unique_l[[i]])) {
-      unique_l2[[i]][j] <- rsm::val2code(eval(parse(text = paste("c(",names(data)[i], "=", as.character(unique_l[[i]][j]),")", sep = ""))), codings = rsm::codings(rsm))
+
+  tmp <- vector("list", dim(possibleCodedCombinationsOfPredictors)[1])
+  for (i in seq_len(dim(possibleCodedCombinationsOfPredictors)[1])) {
+    tmp[[i]] <- optim(as.vector(possibleCodedCombinationsOfPredictors[i,]), rsmOpt_2, dObject = desire_final,
+                      rsm = rsm, space = "square", control = list(fnscale = -1))
+    if (i == 1) {
+      bestCombinedDesirability <- tmp[[i]]
+    }else {
+      if (tmp[[i]][["value"]] > bestCombinedDesirability[["value"]])
+        bestCombinedDesirability <- tmp[[i]]
     }
   }
-  comb2 <- expand.grid(unique_l2)
-  # print(comb)
-  # print(comb2)
 
-  coef <- summary(rsm)[[4]][,1]
 
+
+  if (is.null(jaspResults[["desirability_table"]])) {
+    desirability_table <- createJaspContainer()
+    jaspResults[["desirability_table"]] <- desirability_table
+    desirability_table$dependOn(options = c("rsmMin", "rsmMax", "rsmVariables", "rsmTar"))
+  } else {
+     desirability_table <- jaspResults[["desirability_table"]]
+  }
+
+
+  if (is.null(desirability_table[["predictor_value"]])) {
+    predictor_value <- createJaspTable(title = "Predictor Values")
+    desirability_table[["predictor_value"]] <- predictor_value
+  }else {
+    predictor_value <- desirability_table[["predictor_value"]]
+  }
+
+  if (is.null(desirability_table[["value"]])) {
+    value <- createJaspTable(title = "Overall Desirability")
+    value$addColumnInfo(name = "Value", title = "Value")
+    desirability_table[["value"]] <- value
+  }else {
+    value <- desirability_table[["value"]]
+  }
```

Comment:
You're creating a *container*, but you call it desirability_*table*. Next, it looks like you enter and retrieve tables into other tables. That's a bit confusing, at least to me.